homeassistant/components/zwave_js/humidifier.py

```diff
@@ -28,6 +33,43 @@
 from .entity import ZWaveBaseEntity
 
 
+@dataclass
+class ZwaveHumidifierEntityDescription(HumidifierEntityDescription):
+    """A class that describes the humidifier or dehumidifier entity."""
+
+    # The "on" control mode for this entity, e.g. HUMIDIFY for humidifier
+    on_mode: HumidityControlMode | None = None
+
+    # The "on" control mode for the inverse entity, e.g. DEHUMIDIFY for humidifier
+    inverse_mode: HumidityControlMode | None = None
+
+    # The setpoint type controlled by this entity
+    setpoint_type: HumidityControlSetpointType | None = None
+
+    # Suffix added to name and unique id
+    name_suffix: str | None = None
+
+
+HUMIDIFIER_ENTITY_DESCRIPTION = ZwaveHumidifierEntityDescription(
+    key="0",
+    device_class=HumidifierDeviceClass.HUMIDIFIER,
+    on_mode=HumidityControlMode.HUMIDIFY,
+    inverse_mode=HumidityControlMode.DEHUMIDIFY,
+    setpoint_type=HumidityControlSetpointType.HUMIDIFIER,
+    name_suffix="humidifier",
+)
+
+
+DEHUMIDIFIER_ENTITY_DESCRIPTION = ZwaveHumidifierEntityDescription(
+    key="1",
+    device_class=HumidifierDeviceClass.DEHUMIDIFIER,
+    on_mode=HumidityControlMode.DEHUMIDIFY,
```

Comment:
Are some of the custom attributes required? In that case it's better to create a mixin dataclass for the required attributes so they can be typed correctly as required instead of optional.

Reply:
Updated this as well, sorry I haven't done any typed python before, I was not familiar with dataclasses.